### PR TITLE
{DO NOT MERGE}(MODULES-2709) exploratory

### PR DESCRIPTION
**Pull request by ferventcoder**

### CODE REVIEW

File: lib/puppet_x/puppetlabs/powershell_manager.rb [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+              stdout_monitor_thread = Thread.new do
+                while !@stdout.eof? do
+                  puts "monitor thread - entering stdout_lock synchronize"
+                  @stdout_lock.synchronize do
+                    begin
+                      if !proc_thread.alive?
+                        puts "process has exited so stdout is bad"
+                        raise "Process has exited, no stdout to give"
+                      end
```
Comment (ferventcoder) [inline lib/puppet_x/puppetlabs/powershell_manager.rb]: @Iristyle you want to incorporate this bit back into the original.

Comment (ferventcoder) [inline lib/puppet_x/puppetlabs/powershell_manager.rb]: This is the wrong location. wait one.

File: lib/puppet_x/puppetlabs/powershell_manager.rb [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+                  puts "monitor thread - entering stdout_lock synchronize"
+                  @stdout_lock.synchronize do
+                    begin
+                      # this is necessary to ensure all output captured
+                      @stdout.flush
+                      while self.class.is_readable?(@stdout) do
+                        if !proc_thread.alive?
+                          puts "process has exited so stdout is bad"
+                          raise "Process has exited, no stdout to give"
+                        end
```
Comment (ferventcoder) [inline lib/puppet_x/puppetlabs/powershell_manager.rb]: @Iristyle this is the right location for checking.

File: spec/integration/puppet_x/puppetlabs/powershell_manager_spec.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
+#! /usr/bin/env ruby
+require 'spec_helper'
+require 'puppet/type'
+require 'puppet_x/puppetlabs/powershell_manager'
+
+describe PuppetX::Dsc::PowerShellManager, :if => Puppet::Util::Platform.windows? do
+
+  let (:powershell) {  }
+
+  let (:manager) { PuppetX::Dsc::PowerShellManager.instance("powershell.exe -NoProfile -NonInteractive -NoLogo -ExecutionPolicy Bypass -Command -") }
```
Comment (ferventcoder) [inline spec/integration/puppet_x/puppetlabs/powershell_manager_spec.rb]: @jpogran you likely want to incorporate the argument fixes from this

File: lib/puppet_x/puppetlabs/powershell_manager.rb (hunk 2 above)
Comment (ferventcoder) [inline lib/puppet_x/puppetlabs/powershell_manager.rb]: Of course this isn't quite the right message yet, still rough as I just determined what the issue was.

**Comment by Iristyle** [comment]
So @jpogran / @ferventcoder I went in a different direction to simplify things and remove the threading - see my updated PR that uses `Process.spawn` and is 40 lines shorter than what's here in #165 .

**Comment by ferventcoder** [comment]
@Iristyle I sent up a PR to provide fixes for an exited process we are trying to read stdout off of, but the PR only has questions at the moment.

**Comment by Iristyle** [comment]
Verified #165 is the way to go, so closing this...

**Comment by ferventcoder** [comment]
Of course something similar may need to be looked at when used in general. But for now :+1:

**Pull request closed without merging**